src/hdmf/common/table.py

```diff
@@ -294,6 +294,9 @@ def __init__(self, **kwargs):  # noqa: C901
 
         self.__df_cols = [self.id] + [col_dict[name] for name in self.colnames]
         self.__colids = {name: i+1 for i, name in enumerate(self.colnames)}
+        self._init_class_columns()
+
+    def _init_class_columns(self):
```

Comment:
Note, this will not fix the issue. The single ``_`` makes the function private (i.e., Python will mangle the name), which makes it awkward to call the function from child classes. I'd suggest to use a double ``__`` instead.

Reply:
I thought double `__` makes the function private with Python name mangling and `_` is how you signal that it should only be used internally and by subclasses (protected).

https://stackoverflow.com/questions/1301346/what-is-the-meaning-of-a-single-and-a-double-underscore-before-an-object-name/1301369

Reply:
Calling from a subclass works fine in my tests: https://github.com/NeurodataWithoutBorders/pynwb/pull/1205/files#diff-edc3e1a40efc7c55019dfdfbc9d9cc9cR136-R142